stdlib/_markupbase.pyi

```diff
@@ -1,6 +1,14 @@
+import sys
+
 class ParserBase:
     def __init__(self) -> None: ...
-    def error(self, message: str) -> None: ...
     def reset(self) -> None: ...
     def getpos(self) -> tuple[int, int]: ...
     def unknown_decl(self, data: str) -> None: ...
+    def parse_comment(self, i: int, report: int = ...) -> int: ...  # undocumented
+    def parse_declaration(self, i: int) -> int: ...  # undocumented
+    def parse_marked_section(self, i: int, report: int = ...) -> int: ...  # undocumented
+    def updatepos(self, i: int, j: int) -> int: ...  # undocumented
+    if sys.version_info < (3, 10):
+        # Removed from ParserBase: https://bugs.python.org/issue31844
+        def error(self, message: str) -> None: ...  # undocumented
```

Comment:
```suggestion
        def error(self, message: str) -> NoReturn: ...  # undocumented
```

Since this always raises `NotImplementedError`, it might be better to have `typing.NoReturn` as the return type.

Reply:
This is intended to be overrided in a subclass, so it would be better to set this to whatever subclasses are supposed to do. You can't override a `-> NoReturn` method with a `-> None` method, for example. `NoReturn` might still be correct, if subclasses are expected to always raise an exception in `error()`.

Reply:
True. But `HTMLParser` never actually overrode it, if you look at the BPO issue. So I wonder if it should even be included in the stub for `ParserBase`.

Reply:
After looking at the code in `_markupbase` a bit more, I now agree that `-> NoReturn` is a good idea. For example, this code assumes that `error()` raises an exception, because otherwise the `i+4` passed to `.search()` is meaningless:

```
    def parse_comment(self, i, report=1):
        rawdata = self.rawdata
        if rawdata[i:i+4] != '<!--':
            self.error('unexpected call to parse_comment()')
        match = _commentclose.search(rawdata, i+4)
        ...
```

Reply:
Just to confirm we're happy with keeping error method on ParserBase? (I've switched to `NoReturn`)

Reply:
We can keep it, but it doesn't really matter much, as it appears to be a legacy thing that nobody is using anyway.

Reply:
^agreed